src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,505 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
-            dialogLoading: true,
-            customDiskVerificationFailed: false,
-            customDiskVerificationMessage: null,
-            verificationInProgress: false,
-            file: null,
-        };
-        this.onValueChanged = this.onValueChanged.bind(this);
-        this.dialogErrorSet = this.dialogErrorSet.bind(this);
-        this.onAddClicked = this.onAddClicked.bind(this);
-        this.onInsertClicked = this.onInsertClicked.bind(this);
-        this.existingVolumeNameDelta = this.existingVolumeNameDelta.bind(this);
-        this.validateParams = this.validateParams.bind(this);
-    }
-
-    get initialState() {
-        const { vm, vms, isMediaInsertion } = this.props;
-        const storagePools = getVmStoragePools(vm);
-        const defaultBus = 'virtio';
-        const existingTargets = Object.getOwnPropertyNames(vm.disks);
-        const availableTarget = getNextAvailableTarget(existingTargets, defaultBus);
-        const sortFunction = (poolA, poolB) => poolA.name.localeCompare(poolB.name);
-        let defaultPool;
-        if (storagePools.length > 0)
-            defaultPool = storagePools
-                    .map(pool => ({ name: pool.name, type: pool.type }))
-                    .sort(sortFunction)[0];
-
-        return {
-            storagePools,
-            vm,
-            vms,
-            file: "",
-            device: "disk",
-            storagePoolName: defaultPool && defaultPool.name,
-            storagePoolType: defaultPool && defaultPool.type,
-            mode: isMediaInsertion ? CUSTOM_PATH : CREATE_NEW,
-            volumeName: "",
-            existingVolumeName: undefined,
-            size: 1,
-            unit: units.GiB.name,
-            format: defaultPool && getDefaultVolumeFormat(defaultPool),
-            target: availableTarget,
-            permanent: !domainIsRunning(vm.state), // default true for a down VM; for a running domain, the disk is attached tentatively only
-            hotplug: domainIsRunning(vm.state), // must be kept false for a down VM; the value is not being changed by user
-            addDiskInProgress: false,
-            cacheMode: 'default',
-            busType: defaultBus,
-            updateDisks: false,
-            serial: "",
-        };
-    }
+const sortFunction = (poolA, poolB) => poolA.name.localeCompare(poolB.name);
+
+export const AddDiskModalBody = ({ disk, idPrefix, isMediaInsertion, vm, vms, supportedDiskBusTypes }) => {
+    const [customDiskVerificationFailed, setCustomDiskVerificationFailed] = useState(false);
+    const [customDiskVerificationMessage, setCustomDiskVerificationMessage] = useState(null);
+    const [dialogError, setDialogError] = useState(null);
+    const [dialogErrorDetail, setDialogErrorDetail] = useState(null);
+    const [dialogLoading, setDialogLoading] = useState(true);
+    const [diskParams, setDiskParams] = useState({});
+    const [mode, setMode] = useState(isMediaInsertion ? CUSTOM_PATH : CREATE_NEW);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231116-154931-f2ab7bbc-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)